### PR TITLE
Madeleine add bbox filter

### PR DESCRIPTION
**Pull request by GeOdin**

### CODE REVIEW

**Comment by jackieleng** [comment]
nice

File: lizard_viewer.py [hunk 1]
```diff
@@ -242,12 +247,35 @@ def show_data(self, public_or_private):
             "Downloading {}...".format(asset_type))
         # Get a list with JSONs containing the data from the Lizard API
         payload = {"page_size": 100}
+        # Get the selected public or private area
+        if public_or_private is "private":
```
Comment (jackieleng) [inline lizard_viewer.py]: `is` compares references, use `==` (compares values)

Comment (jackieleng) [reply lizard_viewer.py]: use `==` here also

File: dockwidget.py [hunk 2]
```diff
@@ -63,6 +64,12 @@ def add_datatypes_to_combobox(self, asset_types):
             self.data_type_combobox_private.addItem(asset_type)
             self.data_type_combobox_public.addItem(asset_type)
 
+    def add_areafilters_to_combobox(self):
```
Comment (jackieleng) [inline dockwidget.py:68]: Is there a reason this is dynamically set and not via .iu file?

File: lizard_viewer.py [hunk 3]
```diff
@@ -242,12 +247,35 @@ def show_data(self, public_or_private):
...
+                str(lat1), str(lon1), str(lat2), str(lon2)])
+        # Get the data
         list_of_assets = get_data(asset_type, payload)
-        # Create a new vector layer
-        self.layer = create_layer(asset_type, list_of_assets)
-        # Set the status bar text
-        self.dockwidget.set_all_status_bars_text(
-            "{} downloaded.".format(asset_type.capitalize()))
+        # Create a new layer with the asset data
+        if list_of_assets != []:
```
Comment (jackieleng) [inline lizard_viewer.py]: It is more pythonic to do `if list_of_assets:`

File: lizard_viewer.py [hunk 4]
```diff
@@ -242,12 +247,35 @@ def show_data(self, public_or_private):
...
         payload = {"page_size": 100}
+        # Get the selected public or private area
+        if public_or_private is "private":
+            area_combobox = self.dockwidget.area_combobox_private
+        else:
+            area_combobox = self.dockwidget.area_combobox_public
+        area_index = area_combobox.currentIndex()
+        area_type = AREA_FILTERS[area_index]
+        # Add the bbox to the payload if 'Current view' is selected
+        if area_type == "Current view":
```
Comment (jackieleng) [inline lizard_viewer.py]: `"Current view"` is still kinda a magic string here, I think... 
Is probably okay for now, but it's maybe an idea to put this section into a function.

File: utils/get_data.py [hunk 5]
```diff
@@ -17,11 +21,37 @@ def get_data(asset_type, payload):
...
     response = urllib2.urlopen(url)
     r = json.load(response)
     results = r["results"]
-    # count = r["count"]
 
     # Return the results
     return results
+
+
+def get_bbox(iface):
```
Comment (jackieleng) [inline utils/get_data.py]: What is the reason that this bbox function is in this module, while the other bbox related functions are in the geometry.py module?

**Review by jackieleng — CHANGES_REQUESTED** [review]

File: lizard_viewer.py [hunk 6]
```diff
@@ -242,12 +247,35 @@ def show_data(self, public_or_private):
             "Downloading {}...".format(asset_type))
         # Get a list with JSONs containing the data from the Lizard API
         payload = {"page_size": 100}
+        # Get the selected public or private area
+        if public_or_private is "private":
+            area_combobox = self.dockwidget.area_combobox_private
+        else:
+            area_combobox = self.dockwidget.area_combobox_public
+        area_index = area_combobox.currentIndex()
```
Comment (jackieleng) [inline lizard_viewer.py]: Just a thought, `currentText()` gets the text right? Wouldn't that be more direct?

File: utils/get_data.py (hunk 5 above)
Comment (GeOdin) [reply utils/get_data.py]: I was in doubt whether to put it in the get_data module (getting the bbox as a form of getting data) or the geometry module.
The fact that you comment on this, shows that the geometry module is probably more logical.

File: lizard_viewer.py (hunk 6 above)
Comment (GeOdin) [reply lizard_viewer.py]: Good idea!

**Comment by jackieleng** [comment]
:+1:

**Pull request merged**